reframe/utility/__init__.py

```diff
@@ -1,23 +1,76 @@
+import importlib
 import os
-import importlib.util
+import re
+import sys
 
 
-def import_module_from_file(filename):
-    filename = os.path.expandvars(filename)
-
-    # Figure out a reasonable module name
-    # FIXME: we are not treating specially `__init__.py`
+def _get_module_name(filename):
     barename, _ = os.path.splitext(filename)
+    if os.path.basename(filename) == '__init__.py':
+        barename = os.path.dirname(filename)
+
     if os.path.isabs(barename):
         module_name = os.path.basename(barename)
     else:
-        module_name = barename.replace('/', '.')
+        module_name = barename.replace(os.sep, '.')
+
+    return module_name
+
+
+def _do_import_module_from_file(filename, module_name=None):
+    module_name = module_name or _get_module_name(filename)
+    if module_name in sys.modules:
+        return sys.modules[module_name]
 
     spec = importlib.util.spec_from_file_location(module_name, filename)
     if spec is None:
         raise ImportError("No module named '%s'" % module_name,
                           name=module_name, path=filename)
 
     module = importlib.util.module_from_spec(spec)
+    sys.modules[module_name] = module
     spec.loader.exec_module(module)
     return module
+
+
+def import_module_from_file(filename):
+    """Import module from file."""
+
+    filename = os.path.normpath(os.path.expandvars(filename))
+    if os.path.isdir(filename):
+        filename = os.path.join(filename, '__init__.py')
+
+    module_name = _get_module_name(filename)
+    if os.path.isabs(filename):
+        return _do_import_module_from_file(filename, module_name)
+
+    return importlib.import_module(module_name)
+
+
+def decamelize(s):
+    """Decamelize the string ``s``.
+
+    For example, ``MyBaseClass`` will be converted to ``my_base_class``.
+    """
+
+    if not isinstance(s, str):
+        raise TypeError('decamelize() requires a string argument')
+
+    if not s:
+        return ''
+
+    return re.sub(r'([a-z])([A-Z])', r'\1_\2', s).lower()
```

Comment:
Is it enough to just call `s.lower()`?

Reply:
We finally don't use this function. I have kept it though, because it's a perhaps useful utility.